File: C++/Count Smaller elements.cpp
class Solution {
  public:
    vector<int> constructLowerArray(vector<int> &arr) {
        int n = arr.size();
        vector <int> ans;
        vector <int> temp = arr;
        sort(temp.begin(),temp.end());
        for(int i=0;i<n;i++){
            int index = lower_bound(temp.begin(),temp.end(),arr[i])-temp.begin();
            ans.push_back(index);
            temp.erase(temp.begin()+index);
        }
        return ans;
    }
};

File: JAVA/Count Smaller elements.java
class Solution {
    int[] constructLowerArray(int[] arr) {
        int res[]=new int[arr.length];
        for(int i=0;i<arr.length-1;i++){
            int c=0;
            for(int j=i+1;j<arr.length;j++){
                if(arr[i]>arr[j]){
                    c++;
                }
            }
            res[i]=c;
        }
        return res;
    }
}
